### ISSUE TITLE
Issues with encoder parameters

### ISSUE THREAD

**Issue by Jakobovski** [blackecho/Deep-Learning-TensorFlow#9]
I am running into issues with the encoder parameters.

running 

`encoder = StackedDeepAutoencoder(dae_layers=[60,30])`

raises an IndexError

```
IndexError                                Traceback (most recent call last)
<ipython-input-90-77a8bc4b22b1> in <module>()
----> 1 encoder = StackedDeepAutoencoder(dae_layers=[60,30])
      2 output = encoder.pretrain(train_img)

/python2.7/site-packages/yadlt/models/autoencoder_models/stacked_deep_autoencoder.pyc in __init__(self, dae_layers, model_name, main_dir, models_dir, data_dir, summary_dir, dae_enc_act_func, dae_dec_act_func, dae_loss_func, dae_num_epochs, dae_batch_size, dataset, dae_opt, dae_learning_rate, momentum, finetune_dropout, dae_corr_type, dae_corr_frac, verbose, finetune_loss_func, finetune_enc_act_func, tied_weights, finetune_dec_act_func, dae_l2reg, finetune_batch_size, do_pretrain, finetune_opt, finetune_learning_rate, finetune_num_epochs)
     72                 n_components=layer, main_dir=self.main_dir, model_name=self.model_name + '-' + dae_str,
     73                 models_dir=os.path.join(self.models_dir, dae_str), data_dir=os.path.join(self.data_dir, dae_str),  summary_dir=os.path.join(self.tf_summary_dir, dae_str),
---> 74                 enc_act_func=dae_enc_act_func[l], dec_act_func=dae_dec_act_func[l], loss_func=dae_loss_func[l],
     75                 opt=dae_opt[l], learning_rate=dae_learning_rate[l], l2reg=dae_l2reg,
     76                 momentum=self.momentum, corr_type=dae_corr_type, corr_frac=dae_corr_frac,

IndexError: list index out of range
```

There are a couple of issues:

1.  Why is `index 1` being used from `dae_enc_act_func`? Shouldn't it use `index 0`? This will fix the problem.
2. Why are the `__init__` parameters initialized as lists in the first place? Wouldn't it make more sense to have them as strings?
3. If we are going to have them as a list then they should be of the form `[tanh]` not  `list(['tanh'])`.

**Comment by Jakobovski** [comment]
nevermind. I see I was reading it wrong. Its is `l` not `1`

**Comment by blackecho** [comment]
The parameters are initialized as lists so that one can use different parameters for different autoencoders, for example you can use `dae_learning_rate=[0.01, 0.001]` to use a learning rate of 0.01 for the first layer and 0.001 for the second layer. The `IndexError` is because autoencoder parameters should be of the same length of the number of layers.
In the command line utility, if you give `--dae_learning_rate 0.01` for a three layers autoencoder, it uses 0.01 for all the layers, creating the list [0.01, 0.01, 0.01].
I should add that functionality also to the "non-command line" repo.

The reason I use `list(['tanh'])` instead of `['tanh']` is just because pycharm raises a warning with `['tanh']` without `list()` :smile:

**Comment by blackecho** [comment]
I just published a new version ('0.0.1rc2') where the parameters not setted for all layers will be used for all the layers. It shouldn't give `IndexError` now.
